types_test.go

```diff
@@ -82,3 +83,40 @@ func TestSearchRequest_validate(t *testing.T) {
 		require.Equal(t, "custom", sr.Hybrid.Embedder)
 	})
 }
+
+func TestDocumentsQuery_JSONSerialization(t *testing.T) {
```

Comment:
We need to write the test in the integration suite, not here. Please check the integration tests.

Please write a table-driven test here to verify the "get documents with query" logic.

https://github.com/meilisearch/meilisearch-go/blob/416cfe49d1d8044879f257c38a95bb728c1095c2/integration/index_document_test.go#L11-L42